src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/BuyRevealedMine.java

```diff
@@ -0,0 +1,301 @@
+            
+            Log.DEBUG("User "+buyer.getName()+" has bought card ("+boughtCard.getUuid()+")");
+            Log.DEBUG("New card drawn "+cardDrawn.getUuid());
+
+            ResponseData responseData = new ResponseData(
+                new BuyerDataResponse(
+                    buyer.getUuid(), 
+                    new TokensDataResponse(
+                        buyer.getTokenCount(TokenType.RUBY), 
+                        buyer.getTokenCount(TokenType.EMERALD), 
+                        buyer.getTokenCount(TokenType.SAPPHIRE), 
+                        buyer.getTokenCount(TokenType.DIAMOND), 
+                        buyer.getTokenCount(TokenType.ONYX), 
+                        buyer.getTokenCount(TokenType.GOLD_JOKER)
+                    ), 
+                    boughtCard.getUuid()
+                ), 
+                new CardDataResponse(
+                    cardDrawn.getUuid(),
+                    cardDrawn.getCardTier(),
+                    cardDrawn.getAdditionalToken(),
+                    cardDrawn.getPoints(),
+                    cardDrawn.getCost(TokenType.RUBY),
+                    cardDrawn.getCost(TokenType.EMERALD),
+                    cardDrawn.getCost(TokenType.SAPPHIRE),
+                    cardDrawn.getCost(TokenType.DIAMOND),
+                    cardDrawn.getCost(TokenType.ONYX))
+            );
+
+            ArrayList<User> players = room.getAllUsers();
+            ServerMessage serverMessage = new ServerMessage(
+                userMessage.getContextId(), 
+                ServerMessageType.BUY_REVEALED_MINE_ANNOUNCEMENT, 
+                Result.OK, 
+                responseData
+            );
+
+            for(User player : players){
+                messenger.addMessageToSend(player.getConnectionHashCode(), serverMessage);        
+            }
+
+
+        }catch (Exception e){
+            ErrorResponse errorResponse = new ErrorResponse(
+                Result.FAILURE, 
+                e.getMessage(), 
+                ServerMessageType.BUY_REVEALED_MINE_RESPONSE, 
+                userMessage.getContextId().toString()
+            );
+            messenger.addMessageToSend(connectionHashCode, errorResponse);
+        }
+    }
+
+    private void validateData(DataDTO dataDTO,Database database) throws 
+    UserNotAMemberException, UserAlreadyInRoomException, RoomDoesntExistException, UserTurnException, CardDoesntExistException{
+        
+        User player = database.getUser(dataDTO.userDTO.uuid);
+        
+        //Check if user exits    
+        if(player == null){
+            //TO DO 
+        }
+
+        Room room = database.getRoomWithUser(player.getUuid());
+
+        // Check if room exists whose user is a member of
+        if(room == null){
+            throw new RoomDoesntExistException("Room does not exist whose user is a member of");
+        }
+
+        Game game = room.getGame();
+
+        //Check if user is in game
+        if(game==null){
```

Comment:
Never happening action. Can be removed

Reply:
How is that never happening? Isnt it that server can get any type of request even with existing room and user that belongs to but not in game?

Reply:
Well, if we look at this in that way, then you are right. Practically it won't be possible if everything works right, since user won't see the card on the table if the game is null. Lets use your implementation